Makefile

```diff
@@ -78,6 +78,8 @@ build/llvm.BUILT:
 		-DCLANG_LINKS_TO_CREATE="$(call join-with,;,$(CLANG_LINKS_TO_CREATE))" \
 		-DCMAKE_BUILD_TYPE=MinSizeRel \
 		-DLLVM_ENABLE_TERMINFO=OFF \
+		-DLLVM_LINK_LLVM_DYLIB=ON \
```

Comment:
Sounds good to me if it works.   I assume the `rpath` stuff in the binaries is setup so that they can find the `.so` files relative to themselves?

Reply:
I believe so yeah, `bin/clang` has:

```
$ readelf -a ./bin/clang | rg -i runpath
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../lib]
 ```
 
 I wasn't certain for other platforms so I was relying on CI here during `make check` to fail if there was an issue